### PR TITLE
Allow vault ssh to accept ssh commands in any ssh compatible format

### PR DESCRIPTION
**Pull request by Crazybus**
Previously vault ssh required ssh commands to be in the format `username@hostname <flags> command`. While this works just fine for human users this breaks a lot of automation workflows and is not compatible with the options that the ssh client supports.

Motivation

We currently run ansible which uses vault ssh to connect to hosts. Ansible generates ssh commands with the format `ssh <flags> -o User=username hostname command`. While this is a valid ssh command it currently breaks with vault because vault expects the format to be `username@hostname`. To work around this we currently use a wrapper script to parse the correct username being set by ansible and translate this into a vault ssh compatible `username@hostname` format

Changes

* You can now specify arguments in any order that ssh client allows. All arguments are passed directly to the ssh command and the format isn't modified in any way.
* The username and port are parsed from the specified ssh command. It will accept all of the options supported by the ssh command and also will properly prefer `-p` and `user@` if both options are specified.
* The ssh port is only added from the vault credentials if it hasn't been specified on the command line

### CODE REVIEW

**Comment by jefferai** [comment]
This seems to revert #4673 which you yourself created (to use the hostname instead of IP) -- is that what you meant to do?

**Comment by Crazybus** [comment]
@jefferai Yes. Before vault was replacing the original ssh command and setting it to be `username+"@"+ip`. Changing it to `hostname` was a nice quick fix. However with these changes this fix is no longer needed since the original command is passed through directly to ssh without any modifications (only extra arguments are added as needed).

**Comment by Crazybus** [comment]
Let me add a few more examples in case that didn't clear things up.

In the examples my hostname is `localhost` which resolves to `127.0.0.1` and my current user is `mick`.

Before #4673 
```
$ vault ssh -role bastion -mode otp localhost
ssh mick@127.0.0.1 -p 22
```

After #4673 
```
$ vault ssh -role bastion -mode otp localhost
ssh mick@localhost -p 22
```

Now 
```
$ vault ssh -role bastion -mode otp localhost
ssh localhost -p 22
```
But also any form of username/port specification allowed by ssh also works as expected. The `-p` arg is only appended with the port value from the creds api if it wasn't specified in the original ssh command. 
```
$ vault ssh -role bastion -mode otp -- -o User=jeff -o Port=2222 localhost
ssh -o User=jeff -o Port=2222 localhost
```

**Comment by Crazybus** [comment]
@jefferai I just learned why that `go test -race` flag is so useful :)

```
command/ssh_test.go:143: Errorf format %q reads arg #1, but call has
only 0 args
```

Which I have now fixed in https://github.com/hashicorp/vault/pull/4710/commits/942fd2dd58c6389becef1be6b9564e5f28a1e0ab

File: command/ssh.go [hunk 1]
```diff
@@ -460,10 +470,6 @@ func (c *SSHCommand) handleTypeCA(username, hostname, ip string, sshArgs []strin
 		)
 	}
 
-	args = append(args,
```
Comment (calvn) [inline command/ssh.go]: Why is this removed entirely? We don't need to pass in the username, IP, and port to the SSH command explicitly any more?

Comment (Crazybus) [reply command/ssh.go]: Correct. My comment here should clear things up for you: https://github.com/hashicorp/vault/pull/4710#issuecomment-395293343

The [current behaviour](https://github.com/Crazybus/vault/blob/2b7440524e2ff9d384a7aa61ff148dec2a018a14/command/ssh.go#L253-L257) was to expect username@hostname as the first argument. This was then added to the actual ssh command with username@hostname plus the remaining ssh arguments. This PR removes that limitation of needing to use username@hostname and instead allows you to use any valid ssh command so that vault doesn't need to modify the actual command.

**Comment by jefferai** [comment]
Thanks!
